AK/FlyString.cpp

```diff
@@ -137,7 +96,7 @@ bool FlyString::operator==(FlyString const& other) const
 
 bool FlyString::operator==(String const& other) const
 {
-    if (m_data == other.to_fly_string_data({}))
+    if (m_data == other)
```

Comment:
I'm going to reopen @alimpfard's concerns here; this decays FlyString <=> String comparisons to full string comparisons. And yes, comparing the string data is not possible since non-interned strings may have arbitrary string data pointers.

Reply:
Not present in the end.